torchgeo/datasets/cropharvest.py

```diff
@@ -111,15 +111,7 @@ def __init__(
 
         Raises:
             DatasetNotFoundError: If dataset is not found and *download* is False.
-            ImportError: If h5py is not installed
         """
-        try:
```

Comment:
No need to fail immediately upon instantiation, h5py is only needed for `__getitem__`

Reply:
I disagree, without h5py instantiation won't be useful? I'd rather find out immediately that my dataset won't work.

Reply:
It does allow you to download/extract the dataset and plot images, just not load images. I get your point though. How long of a delay is there in realizing that the dataset doesn't work if we don't error until getitem?